syntax = "proto3";

package foreverbull.backtest;

option go_package = "github.com/lhjnilsson/foreverbull/pkg/backtest/pb";

import "foreverbull/backtest/backtest.proto";
import "foreverbull/backtest/session.proto";

message ListBacktestsRequest {}

message ListBacktestsResponse {
    repeated Backtest backtests = 1;
}

message CreateBacktestRequest {
    Backtest backtest = 1;
}

message CreateBacktestResponse {
    Backtest backtest = 1;
}

message GetBacktestRequest {
    string name = 1;
}

message GetBacktestResponse {
    string name = 1;
    Backtest backtest = 2;
}

message CreateSessionRequest {
    string backtest_name = 1;
}

message CreateSessionResponse {
    foreverbull.backtest.Session session = 1;
}

message GetSessionRequest {
    string session_id = 1;
}

message GetSessionResponse {
    foreverbull.backtest.Session session = 1;
}

service BacktestServicer {
    rpc ListBacktests(ListBacktestsRequest) returns (ListBacktestsResponse) {}
    rpc CreateBacktest(CreateBacktestRequest) returns (CreateBacktestResponse) {}
    rpc GetBacktest(GetBacktestRequest) returns (GetBacktestResponse) {}
    rpc CreateSession(CreateSessionRequest) returns (CreateSessionResponse) {}
    rpc GetSession(GetSessionRequest) returns (GetSessionResponse) {}
}
